pyfr/integrators/dual/pseudo/pseudosteppers.py

```diff
@@ -160,3 +175,162 @@ def step(self, t, dt, dtau):
 
         # Return the index of the bank containing u(n+1,m+1)
         return r1, r0
+
+class DualEmbeddedPairPseudoStepper(BaseDualPseudoStepper):
+    # Coefficients
+    a = []
+    b = []
+    bhat = []
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # Compute the c and error coeffs
+        self.c = [0.0] + [sum(self.b[:i]) + ai for i, ai in enumerate(self.a)]
+        self.e = [b - bh for b, bh in zip(self.b, self.bhat)]
+
+        self._nstages = len(self.c)
+
+        # Register a kernel to multiply rhs with local pseudo time-step
+        self.backend.pointwise.register(
+            'pyfr.integrators.dual.pseudo.kernels.localdtau'
+        )
+
+        tplargs = dict(ndims=self.system.ndims, nvars=self.system.nvars)
+
+        dtaukern = lambda ename, ele: self.backend.kernel(
+            'localdtau', tplargs=tplargs, dims=[ele.nupts, ele.neles],
+             negdivconf=ele.scal_upts_inb, dtau_upts=self.dtau_upts[ename]
+        )
+
+        self.dtau_upts = {}
+        self.eles_scal_dtau_upts = proxylist([])
+
+        for ele, ename, shape in zip(self.system.ele_map.values(),
```

Comment:
for (ename, ele), shape in zip(self.system.ele_map.items(), self.system.ele_shapes)